app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitItemAdapter.java

```diff
@@ -152,6 +163,11 @@ public void onClick(View v) {
                 public void onClick(View v) {
                     int habitPosition = getAdapterPosition();
                     Habit habitToDelete = _snapshots.get(habitPosition);
+                    try {
+                        deleteHabitEvents(username, habitToDelete);
+                    } catch(Error e) {
+                        Log.d(TAG, "No events to delete");
+                    }
```

Comment:
Just a question here... So if an error occurs (meaning there were no events to delete), what happens? Are we still able to delete a habit with an empty event list? I think you mentioned that empty collections aren't possible, but I could be wrong.

Reply:
Yep! If there's no habits in the list, there's actually nothing to delete. Collections are created whenever at least one item is attached, so if there's no habits in the list, there isn't actually a collection attached to the habit